merlin/util/snakewriter.py

```diff
@@ -83,17 +114,22 @@ def _generate_shell(self) -> str:
 
     def as_string(self) -> str:
         fullString = ('rule %s:\n\tinput: %s\n\toutput: %s\n\tmessage: %s\n\t'
-                      + 'shell: %s\n') \
+                      + 'shell: %s\n\n') \
+                     % (self._analysisTask.get_analysis_name(),
+                        self._generate_done_input(), self._generate_output(),
+                        self._generate_message(),  self._generate_shell())
+        fullString += ('rule %sDone:\n\tinput: %s\n\toutput: %s\n\t' +
+                       'shell: \'touch %s\'\n') \
```

Comment:
I'd prefer if there is a way this file can be created even when it isn't run through snakemake. This would likely have to happen in the ParallelAnalysisTask class. Also, the file indicating that the full task is done doesn't have to be a new type of file but instead could use the standard .done file with no index, such as what would be written if it was a non-parallel analysis task. When each step in the ParallelAnalysisTask finishes, it can check if it is the last step to finish and if so write a .done file without an index (record_analysis_complete with fragmentIndex=None).

Reply:
I'd initially implemented it along these lines, but the only way I found to write it resulted in snakemake complaining about output files appearing that it hadn't expected. In essence, I had the same "done" tasks as I do now, but it just didn't have a shell command, and the output would appear because the last fragment of the parallel analysis task created the overall .done file. It didn't cause it to crash, just issue warnings, but it didn't seem like a great choice because the user may view the warning as significant.

An alternative would be to create an analysis task that checks whether a parallel analysis task has completed, and run this as a dependency for downstream tasks. I liked that this would be pretty clear, but I didn't like the added complexity in constructing an analysis parameters file. 

I'm thinking about another way to do it, but if that doesn't work well I'm inclined to leave it the way I've written it unless you want me to just tolerate the snakemake warnings.